ui/jobs/jobs.css

```diff
@@ -1137,3 +1143,8 @@
     opacity: 0;
   }
 }
+
+/** hide bars except hp-bar and mp-bar */
```

Comment:
This is a little bit of what I'm talking about in terms of "why is this patch so big".  This is both a components/event emitter patch, but also adds additional functionality like checking for pvp zones.  I think you could have added the pvp changes before or after the refactoring patch, and it would have made this patch easier to read because it is smaller.

Reply:
Well, the PvP checking logic is included before, what I do was to change the programmatic bars hiding to using CSS to hide them, which I think is much clear. Although a potential cost is that the logic split out into js and css.